rasa/shared/core/domain.py

```diff
@@ -877,6 +890,39 @@ def as_dict(self) -> Dict[Text, Any]:
             KEY_E2E_ACTIONS: self.action_texts,
         }
 
+    def new_function(self, arg1, arg2, arg3, arg4):
+        """A one line docstring is valid."""
+        pass
+
+    def new_function2(self, arg1, arg2, arg3, arg4):
+        """It's not valid if an arg description misses.
+
+        Args:
+            arg1: Arg1.
+            arg2: Arg2.
+        """
+        pass
+
+    def new_function3(self, arg1, arg2, arg3, arg4):
+        """It's stupidly still valid if there is a `Returns` block.
+
+        Args:
+            arg1: Arg1.
+            arg2: Arg2.
+
+        Returns:
+            Something.
+        """
+        pass
+
+    def persist(self, filename: Union[Text, Path]) -> None:
+        """Write domain to a file."""
+
+        domain_data = self.as_dict()
+        rasa.shared.utils.io.write_yaml(
+            domain_data, filename, should_preserve_key_order=True
+        )
+
```

Comment:
what about a valid docstring that's not in Google style, but, say, in Numpy?

Reply:
I say we only allow one style, for consistency

Reply:
yes but I mean a test (even if it's just for the draft PR stage) that checks this

Reply:
works for me on only if it's at the draft PR stage :) otherwise I don't see the point of testing flake8 lib

Reply:
Wait - should I add one to make sure `pydocstyle` verifies it?

Reply:
could be a good idea just to double check, but it doesn't need to be merged in the codebase in my opinion

Reply:
unfortunately it doesn't fail on docstrings with the wrong convention 😬 Seems they will fix this in version 6 of pydocstyle (see https://github.com/PyCQA/pydocstyle/issues/482#issuecomment-679403107)

Reply:
pushed an example with a numpy docstring btw